src/sagemaker/local/local_session.py

```diff
@@ -0,0 +1,168 @@
+
+    def describe_training_job(self, TrainingJobName):
+        """Describe a local traininig job.
+
+        Args:
+            TrainingJobName (str): Not used in this implmentation.
+
+        Returns: (dict) DescribeTrainingJob Response.
+
+        """
+        response = {'ResourceConfig': {'InstanceCount': self.train_container.instance_count},
+                    'TrainingJobStatus': 'Completed',
+                    'TrainingStartTime': datetime.datetime.now(),
+                    'TrainingEndTime': datetime.datetime.now(),
+                    'ModelArtifacts': {'S3ModelArtifacts': self.s3_model_artifacts}
+                    }
+        return response
+
+    def create_model(self, ModelName, PrimaryContainer, ExecutionRoleArn):
+        self.model_name = ModelName
+        self.primary_container = PrimaryContainer
+        self.role_arn = ExecutionRoleArn
+
+    def describe_endpoint_config(self, EndpointConfigName):
+        if self.created_endpoint:
+            return True
+        else:
+            error_response = {'Error': {'Code': 'ValidationException', 'Message': 'Could not find endpoint'}}
+            raise ClientError(error_response, 'describe_endpoint_config')
+
+    def create_endpoint_config(self, EndpointConfigName, ProductionVariants):
+        self.variants = ProductionVariants
+
+    def describe_endpoint(self, EndpointName):
+        return {'EndpointStatus': 'InService'}
+
+    def create_endpoint(self, EndpointName, EndpointConfigName):
+        instance_type = self.variants[0]['InstanceType']
+        instance_count = self.variants[0]['InitialInstanceCount']
+        self.serve_container = _SageMakerContainer(instance_type, instance_count,
+                                                   self.primary_container['Image'], self.sagemaker_session)
+        self.serve_container.serve(self.primary_container)
+        self.created_endpoint = True
+
+        i = 0
+        http = urllib3.PoolManager()
+        serving_port = 8080
+        if self.sagemaker_session.config and 'local' in self.sagemaker_session.config:
+            serving_port = self.sagemaker_session.config['local'].get('serving_port', 8080)
+        endpoint_url = "http://localhost:%s/ping" % serving_port
+        while True:
+            i += 1
+            if i >= 10:
+                raise RuntimeError("Giving up, endpoint: %s didn't launch correctly" % EndpointName)
+
+            logger.info("Checking if endpoint is up, attempt: %s" % i)
+            try:
+                r = http.request('GET', endpoint_url)
+                if r.status != 200:
+                    logger.info("Container still not up, got: %s" % r.status)
+                else:
+                    return
+            except urllib3.exceptions.RequestError:
+                logger.info("Container still not up")
+
+            time.sleep(1)
+
+    def delete_endpoint(self, EndpointName):
+        self.serve_container.stop_serving()
```

Comment:
serve_container is an instance of ```_SageMakerContainer``` and stop_serving will call invoke ```serve_container.container.down()```. Is it necessary to have these two container classes? Can we review the abstraction here?